resource-dispatcher/configuration/__init__.py

```diff
@@ -1,13 +1,45 @@
 import os
 import yaml
 import sys
+from jinja2 import Template
 
 
 def get_config():
-    if "CONFIG_FILE" not in os.environ:
-        sys.exit("Environment variable CONFIG_FILE must be defined or this application has nothing to do. Exiting.")
-    f = open(os.environ["CONFIG_FILE"], "r")
-    contents = f.read()
-    f.close()
-    configuration = yaml.load(contents, Loader=yaml.FullLoader)
-    return configuration
+    # The system management supervisor `execute_supervised` will not let us get here unless exactly one of these is defined.
+    if "CONFIG_DIR" in os.environ:
+        mode = "dir"
+    elif "CONFIG_FILE" in os.environ:
+        mode = "file"
+
+    template_vars = {"ENV_" + var: os.environ[var] for var in os.environ}
+
+    if mode == "file":
+        f = open(os.environ["CONFIG_FILE"], "r")
+        contents = f.read()
+        f.close()
+        contents_template = Template(contents)
+        templated_contents = contents_template.render(**template_vars)
+        configuration = yaml.load(templated_contents, Loader=yaml.FullLoader)
+        print(f"Loaded configuration file: {f.name}")
+        return configuration
+
+    if mode == "dir":
+        configuration = {
+            "repositories": [],
+            "tasks": []
+        }
+        for file in os.listdir(os.environ["CONFIG_DIR"]):
+            if file.endswith(".yml") or file.endswith(".yaml"):
+                path = os.path.join(os.environ["CONFIG_DIR"], file)
+                f = open(path, "r")
+                contents = f.read()
+                f.close()
+                contents_template = Template(contents)
+                templated_contents = contents_template.render(**template_vars)
+                config_tmp = yaml.load(templated_contents, Loader=yaml.FullLoader)
+                if "repositories" in config_tmp:
+                    configuration["repositories"].extend(config_tmp["repositories"])
+                if "tasks" in config_tmp:
+                    configuration["tasks"].extend(config_tmp["tasks"])
+                print(f"Loaded configuration file: {path}")
```

Comment:
A couple of suggestions that will perhaps help with troubleshooting later on:
1. make sure to indicate that his was in the `dir` mode - i.e.: right now this line is the same as 23 above (yes, it will show the path, but still good to differentiate the actual message IMHO). Perhaps also add a message earlier on (approx line 32) indicating that it is going down the directory path
2. If a file is skipped due to not ending in `.yml` or `.yaml`, perhaps indicate so (?)
3. unless the content is `repositories` or `tasks`, it won't be consumed - perhaps indicate that to the user (catches typos, etc.)

related to #2 - the `file` mode above doesn't do any `.yml` or `.yaml` checks - perhaps it should (to be consistent with directory processing)?